src/base/log.cc

```diff
@@ -148,9 +148,44 @@ void set_logging_dir() {
 std::unique_ptr<Logger> logger(new Logger());
 #endif
 
-void log(const char* const fmt, ...) {
+static const char* to_string(const LogType& type) {
+	switch (type) {
+	case LogType::kInfo:
+		return "Info   ";
+	case LogType::kDebug:
+		return "Debug  ";
+	case LogType::kWarning:
+		return "WARNING";
+	case LogType::kError:
+		return "ERROR  ";
+	default:
+		NEVER_HERE();
+	}
+}
+
+void log_to_stdout(const LogType type, int64_t gametime, const char* const fmt, ...) {
 	assert(logger != nullptr);
 
+	{  // message type
+		char buffer[16];
+		sprintf(buffer, "[%s", to_string(type));
+		logger->log_cstring(buffer);
+	}
+	if (gametime >= 0) {  // timestamp
```

Comment:
Create a new constant for this (numeric_limits::max) and use it in the molog(-1 calls. Time should eventually become an unsigned, so having -1 here introduces a new complication. See https://github.com/widelands/widelands/issues/1561